src/js/components/uploadDetachedFiles/uploadDetachedFileMeta.jsx

```diff
@@ -0,0 +1,458 @@
+import moment from 'moment';
+
+import Navbar from '../SharedComponents/navigation/NavigationComponent.jsx';
+import Footer from '../SharedComponents/FooterComponent.jsx';
+import SubTierAgencyListContainer from '../../containers/SharedContainers/SubTierAgencyListContainer.jsx';
+import ValidateValuesFileContainer from '../../containers/validateData/ValidateValuesFileContainer.jsx';
+import ValidateDataFileContainer from '../../containers/validateData/ValidateDataFileContainer.jsx';
+import UploadDetachedFilesBox from './UploadDetachedFilesBox.jsx';
+import DateRangeWrapper from './DateRangeWrapper.jsx';
+
+import * as Icons from '../SharedComponents/icons/Icons.jsx';
+
+import * as UploadHelper from '../../helpers/uploadHelper.js';
+import * as GenerateFilesHelper from '../../helpers/generateFilesHelper.js';
+import * as ReviewHelper from '../../helpers/reviewHelper.js';
+import { kGlobalConstants } from '../../GlobalConstants.js';
+
+const timerDuration = 5;
+
+export default class UploadDetachedFileMeta extends React.Component {
+	constructor(props) {
+		super(props);
+
+		this.isUnmounted = false;
+
+		this.state = {
+			agency: "",
+			agencyError: false,
+			showDatePicker: false,
+			showUploadFilesBox: false,
+			detachedAward: {
+				startDate: null,
+				endDate: null,
+				error: {
+					show: false,
+					header: '',
+					description: ''
+				},
+				valid: false,
+				status: ""
+			},
+			jobResults: {detached_award: {}},
+			notAllowed: false,
+			errorMessage: "",
+			headerErrors: false,
+			validationFinished: false,
+			error: 0
+		};
+	}
+
+	componentDidMount() {
+		this.isUnmounted = false;
+	}
+
+	componentWillUnmount() {
+		this.isUnmounted = true;
+	}
+
+	handleChange(agency, isValid){
+		// set Sub-Tier Agency and update validity
+		if (agency != '' && isValid) {
+			this.setState({
+				agency: agency,
+				agencyError: false
+			 }, this.agencyChangeComplete);
+		}
+		else {
+			this.setState({
+				agency: '',
+				agencyError: true
+			}, this.agencyChangeComplete);
+		}
+	}
+
+	agencyChangeComplete() {
+		// Display or hide the file upload box
+		if (this.state.agency !== '') {
+			this.setState({
+				showDatePicker: true
+			});
+		} else {
+			this.setState({
+				showDatePicker: false,
+				showUploadFilesBox: false
+			});
+		}
+	}
+
+	handleDateChange(file, date, dateType) {
+		// merge the new date into the file's state without affecting the other keys
+		const newState = Object.assign(this.state[file], {
+			[dateType]: moment(date)
+		});
+
+		this.setState({
+			[file]: newState
+		}, () => {
+			this.validateDates(file);
+		});
+	}
+
+	validateDates(file) {
+		// validate that dates are provided for both fields and the end dates don't come before the start dates
+		let state = "incomplete";
+
+		const dFile = Object.assign({}, this.state[file]);
+
+		// validate the date ranges
+		const start = this.state[file].startDate;
+		const end = this.state[file].endDate;
+		if (start && end) {
```

Comment:
I feel like this could be reduce to just a single if/else with dynamic `dFile.valid` assignments